### ISSUE TITLE
Case sensitivity in tags

### ISSUE THREAD

**Issue by apnadkarni** [tclssg/tclssg#72]
I got the following error when building the site:

```
*** error: UNIQUE constraint failed: input.file ***
Traceback:
UNIQUE constraint failed: input.file
    while executing
"tclssg-db eval {
            INSERT INTO input(file, type, raw, cooked, timestamp)
            SELECT :file2, type, raw, cooked, timestamp FROM input
..."
    (procedure "copy" line 2)
    invoked from within
"db input copy $tagIndexInput $input"
    (procedure "::tclssg::pipeline::60-generate-tag-pages::transform" line 17)
    invoked from within
"${ns}::transform"
    (procedure "run-pipeline-stage" line 6)
    invoked from within
"run-pipeline-stage $ns $files"
    (procedure "compile-website" line 41)
    invoked from within
"compile-website -inputDir $inputDir  -outputDir $outputDir  -debugDir $debugDir  -config $websiteConfig  -plugins $plugins"
    (procedure "build" line 19)
    invoked from within
"tclssg cli command $command  $inputDir  $outputDir  $debugDir  $options"
```

The cause turned out to be tags that differed only in case "windows" and "Windows".

If tags are case-sensitive, then both should be accepted and treated as different values. If case-insensitive, both should be accepted as the same tag. In my case, they were intended to be the same but either treatment would be acceptable in my opinion.

**Comment by apnadkarni** [comment]
As an aside, even an error message disallowing tags differing only in case would be acceptable though not ideal, but the current error output makes it hard for the end user to determine the cause.

**Comment by dbohdan** [comment]
A gnarly error message.  This happens due to an omission in the schema of the internal DB.  The way that literal tags map to tag page filenames means that tags must not only be case-insensitive, but collated by their URL slug, i.e., `=Hello, World!=` and `hello world` must be treated as the same tag (effectively `hello-world`).  I've pushed a fix to `dev` (6fde5d7).
